### ISSUE TITLE
Issue with the inverse problem of an ODE , wrong inferred variable value even when loss is low

### ISSUE THREAD

**Issue by khawlaoue** [lululxvi/deepxde#640]
Hello @lululxvi ,

I have an issue with the inverse problem and I have been trying hard to solve it, I went through the FAQ and tried all the proposed solutions one by one and yet I still can´t figure out the source of the problem.

As I had mentioned before in a previous issue , I have one non linear ODE equation with an unknown A variable. I am trying to infer the value of A and I used the inverse Lorenz system as a basis of my code. At first the problem was with the very big loss values , after reading FAQ , i scaled my input and output and now the loss values are better however the value of A is still wrong.
this is my code:

```
def gen_traindata():
     DP_train = pd.read_csv( "/content/DATA.csv", delimiter=';')
     DP_train.columns=['time','DP']
     t = DP_train['time'].values
     DP = DP_train['DP'].values
     t = t.astype('int32')
     DP=DP.astype('float32')
     t=t.reshape(len(t),1)
     DP=DP.reshape(len(DP),1)
     DP=DP/1000
     t=t/365
     return t, DP

A=dde.Variable(1.0)
E=111*(10**3)
T=98
R=8.314

def Emsley_system(t,y):
    k=A*(10**8)*(tf.math.exp(-E / (R * (T+371))))
    dy_t = tf.gradients(y,t)[0]
    return [ 
             dy_t+k*(y**2)
          
    ]

geom = dde.geometry.TimeDomain(0,6)
ic1 = dde.icbc.IC(geom, lambda X: 1, boundary, component=0)
ob_t,ob_y = gen_traindata()
t_train, t_test, y_train, y_test = ob_t[:len(ob_t)*7//10], ob_t[len(ob_t)*7//10:], ob_y[:len(ob_t)*7//10], ob_y[len(ob_t)*7//10:]

observe_y0 = dde.icbc.PointSetBC(t_train, y_train, component=0)

data = dde.data.PDE(
    geom,
    Emsley_system,
    [ic1, observe_y0],
    #num_domain=100,
    num_boundary=2,
    anchors=t_train,
    num_test=1000
)
net = dde.maps.FNN([1] + [50] * 3 + [1], "swish", "Glorot normal")
model = dde.Model(data, net)
model.compile("adam", lr=0.001,loss='MSE',external_trainable_variables=[A])
variable = dde.callbacks.VariableValue(
     [A], period=600, filename="variables.dat"
 ) 
losshistory, train_state = model.train(epochs=60000,callbacks=[variable])
dde.saveplot(losshistory, train_state, issave=True, isplot=True)

Best model at step 58000:
  train loss: 2.23e-07
  test loss: 2.30e-07
  test metric: []


```
the true value of A is 7.8*(10**8) and the inferred value is never stable and it converges to different values even though it´s the same code and I didnt change anything.

I tried with adam and lbfgsb and still no improvement.

My questions are :
1)why do I get instable predictions?
2) why does the loss still varies even when A value converges ? isn't the loss supposed to converge as well ?
3)how does the algorithm update the value of A ? Sometimes the true value of A comes up and still it keeps incrementing the variable's value and doesnt consider it to be the optimized value .
![image](https://user-images.githubusercontent.com/104361372/165085279-250ec176-b1ef-4707-9be9-13a7463a3158.png)

this is the training dataset :
![image](https://user-images.githubusercontent.com/104361372/165085335-749203e4-cb9d-49b3-b61f-c641840dad48.png)

I kindly hope that you can help me understand the source of the problem and thanks in advance.

**Comment by lululxvi** [comment]
7.8*(10**8) cannot be handled by machine learning.

**Comment by khawlaoue** [comment]
> 7.8*(10**8) cannot be handled by machine learning.

Thank you @lululxvi for your reply , I understand it´s a big value that´s why I added the term *(10**8 ) in the ODE expression in the code so that the inferred value would be in the range of O(1), I am not sure if that fixes it , I  also don't understand why , with different inferred A values , the loss is very loss (in the order of e-7 and it can´t converge to the true value (the one that I created the data with) I am not sure if there's something wrong with my implementation or if that the equation being non linear causes a problem , even though it´s a simple equation with a known solution

if you have a tip or a suggestion , I would really appreciate it,
thanks again

**Comment by lululxvi** [comment]
You cannot have something like 10**8 in the code, otherwise the training is likely unstable. Do a log scale.

**Comment by khawlaoue** [comment]
@lululxvi Thank you so much for your reply ! I am sorry to bother you again but I thought of using the log and when I tried it I assumed that the variable A in the code is actually log(A) and I added exp(A) in the equation but I still didn't get the results I want 
is that what you meant by doing a log scale ? If not , can you please explain to me exactly howto use the log scale in my code ?

Thanks a lot !

A=dde.Variable(5.0) 
E=111*(10**3)
T=98
R=8.314

def Emsley_system(t,y):
    k=np.exp(A)*(np.exp(-E / (R * (T+371))))
    dy_t = tf.gradients(y,t)[0]
    return [ 
             dy_t+k*(y**2)
          
    ]`

**Comment by khawlaoue** [comment]
@lululxvi  Alright ! thanks a lot for your response !
